feature/text_based.py

```diff
@@ -0,0 +1,694 @@
+        # Run greedy heuristic
+        elif optimization_method == "greedy":
+            selected = self._select_greedy()
+        # Run random selection algorithm
+        elif optimization_method == "random":
+            selected = self._select_random(trials=trials)
+        else:
+            raise NotImplementedError(f"{optimization_method} has not been implemented.")
+
+        # Sort selection
+        selected.sort()
+
+        return selected
+
+    def _process_df(self, input_df: pd.DataFrame, categories: List[int], selection_size: int,
+                    optimization_method: str, cost_metric: str) -> NoReturn:
+        """_process_df
+        We have implemented two versions of this function. In the version here, the categories parameter
+        actually refers to the labels (rather than categories) to be covered. It generates features for input data
+        using a specified featurization method, and creates a matrix where each row corresponds to a label and each
+        column corresponds to a feature.
+        The version of code for generating labels from categories has been moved to utils.py.
+
+        Parameters
+        ----------
+        input_df: pd.DataFrame
+            Input data frame with categories and features of content to select from.
+
+        categories: List[int]
+            List of columns in data that contains categories/labels to be covered.
+
+        selection_size: int
+            Number of feature to select.
+
+        optimization_method: str
+            The function need method to prevent performing featurization in
+             the case that optimization method is random(t or unicost) or greedy(unicost)
+
+        cost_metric: str
+            The function need cost metric to prevent performing featurization in
+             the case that cost metric is None or unicost (for random and greedy)
+
+        Returns
+        -------
+        No return
+       --------
+        """
+        self.matrix = categories.to_numpy()
+        self._num_rows, self._num_cols = self.matrix.shape
+        option_map = {
+            ("random", True, "diverse"): lambda: None,
+            ("random", False, "unicost"): lambda: None,
+            ("random", True, "unicost"): lambda: None,
+            ("greedy", False, "unicost"): lambda: None,
+            ("greedy", True, "unicost"): lambda: None,
+            ("exact", True, "unicost"): lambda: None,
+            ("exact", False, "unicost"): lambda: None
+        }
+
+        options = option_map.get((optimization_method, selection_size is not None, cost_metric))
+        if options:
+            options()
+        else:
+            feature_column = self.featurization_method.fit_transform(input_df)
+            self.features = np.array([eval(l) if isinstance(l, str) else l for l in feature_column.tolist()])
+            check_true(len(self.features) == self._num_cols,
+                       f"Process Data Error: features size ({len(self.features)}) "
```

Comment:
these should not be strings.. the second argument is an Exception.. 

That said,  these conditions should NOT occure, correct? The code should never come here. 

We are done with all our validate_args(), validate_fit() functions etc. so by the time the code is deep in the library all should work.. Anytime it does not, it is a because of a bug or sth  --not because expectations are not met. 

Agree that code should not hit these invalid conditions?

Reply:
Yes, I agree that the code should ideally never hit these invalid conditions. However, it might be good to include such checks using raise ValueError as it is clearly indicates that an error has occurred and provides a meaningful error message. 
Please let me know if I need to remove it.